realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -539,4 +632,72 @@ public FieldMetaData(RealmFieldType realmType, boolean defaultNullable) {
             this.defaultNullable = defaultNullable;
         }
     }
+
+    static class DynamicColumnMap implements Map<String, Long> {
```

Comment:
Because this needs to be compatible with the ColumnIndices class. I looked into a rewrite, but it turned out to be a bigger thing so would rather do that outside the scope of this PR